pkgs/development/r-modules/default.nix

```diff
@@ -950,6 +950,18 @@ let
       preConfigure = "patchShebangs configure";
     });
 
+    littler = old.littler.overrideAttrs (attrs: with pkgs; {
+      buildInputs = [ pcre lzma zlib bzip2 icu which ] ++ attrs.buildInputs;
+      propagatedBuildInputs = [ which ] ++ attrs.propagatedBuildInputs;
```

Comment:
Why does this package have to propagate `which`?

Reply:
Because it won't build otherwise. Example error output in case this is left out:
```
** testing if installed package can be loaded
Error: package or namespace load failed for 'littler':
 .onAttach failed in attachNamespace() for 'littler', details:
  call: system(paste(which, shQuote(names[i])), intern = TRUE, ignore.stderr = TRUE)
  error: error in running command
Error: loading failed
```

Taking it one level back it seems that access to `which` is actually a core dependency of R. There are 15+ package overrides present in https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/r-modules/default.nix that put `which` into propagatedBuildInputs on a per-package basis. Perhaps it would even cleaner to move `which` from `buildInputs` to `propagatedBuildInputs` in https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/math/R/default.nix#L18. Thoughts?